src/client/api/container.c

```diff
@@ -51,6 +51,64 @@ daos_cont_create(daos_handle_t poh, const uuid_t uuid, daos_prop_t *cont_prop,
 	return dc_task_schedule(task, true);
 }
 
+int
+daos_cont_create_by_label(daos_handle_t poh, const char *label,
+			  daos_prop_t *cont_prop, uuid_t *uuid,
+			  daos_event_t *ev)
+{
+	size_t			 label_len = 0;
+	daos_prop_t		*label_prop;
+	daos_prop_t		*merged_props = NULL;
+	uuid_t			 cuuid;
+	int			 rc;
+
+	if (label)
+		label_len = strnlen(label, DAOS_PROP_LABEL_MAX_LEN+1);
+	if (!label || (label_len == 0) || (label_len > DAOS_PROP_LABEL_MAX_LEN))
+		return -DER_INVAL;
```

Comment:
I've getting confused with all the PRs in flight but this should be using the new daos_label_is_valid() function presumably?